### PR TITLE
Update M4 Autoconf Files

### PR DESCRIPTION
**Pull request by ejohnstown**
Update the autoconf m4 files, except pthreads which should be its own commit.

### CODE REVIEW

File: m4/ax_check_library.m4 [hunk 1]
```diff
@@ -61,35 +60,36 @@
 #   modified version of the Autoconf Macro, you may extend this special
 #   exception to the GPL to apply to your modified version as well.
 
-#serial 7
+#serial 5
```
Comment (toddouska) [inline m4/ax_check_library.m4]: Is this m4 file going to a previous version?  From version 7 to version 5.  All the other files appear to be moving forward.

Comment (ejohnstown) [reply m4/ax_check_library.m4]: This is one of the files that Brian had made his own updates to (serial 6 and serial 7) for his application. I'm assuming he's returning that file to the actual released version as his changes don't apply to our build process. (I'm am verifying my assumption with him.)

**Comment by drbild** [comment]
Is there a separate PR for updating `ax_pthreads.m4`?  The latest version of it is needed to build on current OS X platforms.

**Comment by ejohnstown** [comment]
@drbild, there isn't a PR for the file `ax_pthreads.m4` yet.

While it is outputting `clang: warning: argument unused during compilation: '-pthread'`, it doesn't look like it is treating it as a warning with respect to `-Werror`. And it looks like, on Sierra with the new ax_pthreads.m4, that libtool isn't adding `-Qunused-arguments` to the flags for linking the dylib.

After doing a fresh clone of the code with this update-m4 branch on my High Sierra machine, and running:

    $ ./autogen.sh && ./configure --disable-silent-rules && make

I'm not seeing the clang pthread warning. Configure didn't add `-pthread -Qunused-arguements` to any of the CFLAGS.

**Comment by drbild** [comment]
@ejohnstown Ok, just curious.  Thanks.

On my Sierra box (and our Travis-CI builds), the current master is treating `clang: warning: argument unused during compilation: '-pthread'` as an error.  Updating my fork to the the latest `ax_pthreads.m4` fixed it.

**Comment by ejohnstown** [comment]
To capture this somewhere, I modified the libtool script that configure produces to add the `-Qunused-arguments` to the `-pthread` option and it built clean on Sierra using the new ax_pthread.m4. Unfortunately, rerunning configure overwrites the libtool script. Libtool builds its own flags and isn't adding the Q argument.

**Comment by ejohnstown** [comment]
I'm updating the ax_pthread.m4 file now. I added a check to see if the `PTHREAD_CFLAGS` had `-Qunused-arguements` and if so prepend it with `-Xcompiler`. The Q option gets passed down to libtool and not filtered out.

**Pull request merged**